apps/services/user-notification/src/app/modules/notifications/dto/notification.dto.ts

```diff
@@ -0,0 +1,173 @@
+import { PageInfoDto, PaginationDto } from '@island.is/nest/pagination'
+
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger'
+import {
+  IsEnum,
+  IsString,
+  IsInt,
+  IsDate,
+  IsArray,
+  ValidateNested,
+  IsOptional,
+} from 'class-validator'
+import { Type } from 'class-transformer'
+
+export enum NotificationStatus {
+  READ = 'read',
+  UNREAD = 'unread',
+}
+
+export class ExtendedPaginationDto extends PaginationDto {
+  @IsOptional()
+  @ApiPropertyOptional({
+    description: 'locale',
+    type: 'string',
+  })
+  @IsString()
+  locale?: string
+}
+
+export class ArgItem {
+  @ApiProperty({ example: 'key1' })
+  @IsString()
+  key!: string
+
+  @ApiProperty({ example: 'value1' })
+  @IsString()
+  value!: string
+}
+
+export class NotificationDto {
+  @ApiProperty({ example: 123 })
+  @IsInt()
+  id!: number
+
+  @ApiProperty({ example: 'uuid-message-id' })
+  @IsString()
+  messageId!: string
+
+  @ApiProperty({ example: 'recipient-uuid' })
+  @IsString()
+  recipient!: string
+
+  @ApiProperty({ example: 'template-uuid' })
+  @IsString()
+  templateId!: string
+
+  @ApiProperty({ type: [ArgItem], example: [{ key: 'key1', value: 'value1' }] })
+  @IsArray()
+  @ValidateNested({ each: true })
+  @Type(() => ArgItem)
+  args!: ArgItem[]
+
+  @ApiProperty({ example: new Date().toISOString() })
+  @IsDate()
+  created!: Date
+
+  @ApiProperty({ example: new Date().toISOString() })
+  @IsDate()
+  updated!: Date
+
+  @ApiProperty({ enum: NotificationStatus, example: NotificationStatus.UNREAD })
+  @IsEnum(NotificationStatus)
+  status!: NotificationStatus
+}
+
+export class RenderedNotificationDto {
+  @ApiProperty({ example: 123 })
+  @IsInt()
+  id!: number
+
+  @ApiProperty({ example: 'uuid-message-id' })
+  @IsString()
+  messageId!: string
+
+  @ApiProperty({ example: 'hnipp' })
+  @IsString()
+  title!: string
+  @ApiProperty({ example: 'hnipp' })
+  @IsString()
+  body!: string
+  @ApiProperty({ example: 'hnipp' })
+  @IsString()
+  dataCopy!: string | null | undefined
+  @ApiProperty({ example: 'hnipp' })
+  @IsString()
+  clickAction!: string | null | undefined
```

Comment:
Can the distinction for these properties be made clearer with differentiated examples?

Reply:
good point - will make this clearer